### ISSUE TITLE
MLCP fails if no "jobId" parameter specified even with trace off

### ISSUE THREAD

**Issue by jkerr5** [marklogic-community/marklogic-data-hub#426]
#### The issue

Short description of the problem:

If I create a gradle mlcp task to run an input flow and do now include a "jobId" transform parameter, mlcp fails with this error:

```
2017-06-12 14:25:54 ERROR ContentWriter:107 - XDMP-AS: (err:XPTY0004) $job-id as xs:string -- Invalid coercion: () as xs:string
2
```
If I have tracing off, should I still have to create a jobId? Also, the examples on the wiki do not include the jobId parameter so it is not clear that you need it to begin with.

#### Tech details

Which Operating System are you using?
OS X

Which version of MarkLogic are you using?
9.0-1.1

Which version of the Data Hub Framework are you using?
1.1.3

Is this a QuickStart UI bug? If so, which browser are you using?
No

#### The devil is in the details...

It looks like this comes from this line in ```/com.marklogic.hub/mlcp-flow-transform.xqy```:
``` 
let $_ := trace:set-job-id(map:get($params, "jobId"))
```
This gets call even if tracing is not enabled. Maybe it should be wrapped in a
```
if (trace:enabled()) then
```

**Comment by txpet** [comment]
I have the same experience. The jobs execute fine when using the data hub portal, but if you use the mlcp command generated, it fails with the message above. I tried to modify the template as suggested above and recompile, but it still does not work. This is a major annoyance since I have hundreds of entities and I would like to automate the process.

**Comment by sashamitrovich** [comment]
Any progress on fixing this? Thanks.

**Comment by paxtonhare** [comment]
it's not so simple to fix because tracing will fire if an exception is caught, even if turned off.

A solid workaround for now is to pass in a jobId.

**Comment by damonfeldman** [comment]
For some reason in quickstart-1.1.3 I got 

12:22:12.387 [pool-1-thread-2] ERROR c.m.contentpump.TransformWriter - QueryException:XDMP-AS: (err:XPTY0004) $transform-option as map:map? -- Invalid coercion: ("Acme%20Medical", "Acme%20Medical", "fooobar-redacted", ...) as map:map?
 
and this issue dissapeared when I added the required JobID in the mlcp command line.

I don't see any way to set the jobId in QuickStart, however. Perhaps we need an "additional params" section on the quickstart GUI for running input flows.
